firmware/CMakeLists.txt

```diff
@@ -19,6 +19,28 @@ elseif("${PLATFORM}" STREQUAL "x86")
     add_custom_target(all_tests)
 endif()
 
+# Generate header file for commit message
+function(commit_info_code_generation
+    HEADER_OUTPUT_PATH
+    SRC_OUTPUT_PATH
+)
+    set(GENERATE_COMMIT_INFO_SCRIPT_PY
+    ${SCRIPTS_DIR}/code_generation/commit_info_gen/src/generate_commit_info.py)
```

Comment:
perhaps set this as a global variable, so it doesn't get regenerated on each run, less side effects

Reply:
or possibly just inline it on L33 (but that would make it slightly less readable)